### DISCUSSION TITLE
Is the macro syntax stable yet?

### DISCUSSION THREAD

**Discussion by SamuraiCrow** [gxi-rs/gxi#20] [Q&A] [answered]
There is a Rust-based operating system called [Theseus OS](https://github.com/theseus-os/theseus/)  that doesn't have a GUI yet.  I was wondering if the GXI backend macro language was stable enough for me to try to create a new optimized grid-based layout structure as a layout engine.

The idea is to macro combine all spans using a JIT to optimize it in place of the transpiler.  Constant folding and inlining optimizations are key to success.  The end execution will queue jobs into a thread pool to resize the GUI sections as items are added and removed, and also as the window is resized.

I want to make GXI the native GUI framework for Theseus.

**Comment by aniketfuryrocks** [comment, marked as answer]
The project is very early in development, so I will strongly discourage the usage of `gxi` anywhere.
